pages/slice/language-reference/slice-grammar.md

```diff
@@ -2,14 +2,581 @@
 title: Slice grammar
 ---
 
+This page describes the grammatical rules, usage, and restrictions of the Slice language.
+
+## Slice elements
+
+### Slice files
+
+The `SliceFile` grammar rule is the parser's entry point.
+Once a file's contents have been [preprocessed], the resulting text is matched against this rule, in its entirety.
+
+While the contents of a Slice file may reference definitions in other files, each file is parsed separately.
+There is no notion of 'including' the contents of one file in another, and errors in one file do not affect the parsing of other files.
+
+A slice file consists of any number of [file attributes][attribute] and at most one [mode statement][mode-statement].
+File attributes can appear before or after (inclusively) the mode statement.
+This is followed by an optional [module declaration][module-declaration], followed by any number of Slice definitions.
+It is illegal for a file to contain Slice definitions without a module declaration.
```

Comment:
I know the intent of my original comment on this first ilegal comment got muddled so for clarity I think this should be changed too.